src/app.ts

```diff
@@ -1,26 +1,65 @@
-// example interfaces that can be use
-// TIP: the types mentioned in the interfaces must be fulfilled in order to solve the problem.
-interface TemperatureReading {
-  time: Date
-  temperature: number
-  city: string
-}
-interface TemperatureSummary {
-  first: number
-  last: number
-  high: number
-  low: number
-  average: number
+import { TemperatureReading, TemperatureSummary } from './interfaces'
+
+type ProccesedReadings = {
+  [city: string]: {
+    [date: string]: number[]
+  }
 }
+let processedReadings: ProccesedReadings = {}
 
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  const processedData = readings.reduce<ProccesedReadings>((data, value) => {
+    const { city, time, temperature } = value
+    const formatedDate = time.toISOString()
+
+    // Validate if city exists or isn't empty
+    if (city.trim() === '' || !city) {
+      throw new Error('Some register has no valid city')
+    }
```

Comment:
What would happen if for instance we're processing 10k data entries? Do you think that throwing an error would cause a negative or positive impact to the data processing?